tlatools/org.lamport.tlatools/test/util/CorpusParser.java

```diff
@@ -346,4 +346,25 @@ public static List<CorpusTestFile> getAndParseCorpusTestFiles(Path toolsRoot) th
 	
 		return corpus;
 	}
+
+	/**
+	 * Gets the TLA+ tools root directory. Requires basedir property was
+	 * provided to the Java VM on test startup.
+	 * 
+	 * In Eclipse, add -Dbasedir=${workspace_loc:tlatools} to the VM
+	 * arguments in the JUnit run configuration.
+	 * 
+	 * @return A Path object pointing to the tools root directory.
+	 */
+	public static Path getToolsRoot() {
+		String baseDir = System.getProperty("basedir");
+		if (null == baseDir) {
+			throw new IllegalArgumentException(
+				"Require basedir Java VM environment variable to be provided"
+				+ " and set to tlatools root."
+			);
+		}
+		
+		return Paths.get(baseDir);
```

Comment:
Use `return Paths.get(CommonTestCase.BASE_DIR)` instead. (You may have to make that field public.)

(Actually, what might be even better is a separate PR that replaces all the basedir stuff in the parse tests with proper Java resources so the tests do not depend on system properties or the current working directory.)

Reply:
Made the changes here. I can pull them out if necessary; it would be a bit annoying with rebase conflicts but either works.